pkg/executor/plugins/rpc.go

```diff
@@ -0,0 +1,177 @@
+package plugins
+
+import (
+	"context"
+	"io"
+	"net/rpc"
+
+	"github.com/bacalhau-project/bacalhau/pkg/bidstrategy"
+	"github.com/bacalhau-project/bacalhau/pkg/model"
+)
+
+type ServerRPC struct {
+	Impl PluggableExecutor
+}
+
+func (s *ServerRPC) IsInstalled(_ IsInstalledRequest, reply *IsInstalledResponse) error {
+	result, err := s.Impl.IsInstalled(context.TODO())
+	if err != nil {
+		return err
+	}
+	reply.Installed = result
+	return nil
+}
+
+func (s *ServerRPC) GetVolumeSize(args GetVolumeSizeRequest, reply *GetVolumeSizeResponse) error {
+	result, err := s.Impl.GetVolumeSize(context.TODO(), args.Spec)
+	if err != nil {
+		return err
+	}
+	reply.Size = result
+	return nil
+}
+
+func (s *ServerRPC) HasStorageLocally(args HasStorageLocallyRequest, reply *HasStorageLocallyResponse) error {
+	result, err := s.Impl.HasStorageLocally(context.TODO(), args.Spec)
+	if err != nil {
+		return err
+	}
+	reply.Has = result
+	return nil
+}
+
+func (s *ServerRPC) ShouldBidBasedOnUsage(args ShouldBidBasedOnUsageRequest, reply *ShouldBidResponse) error {
+	result, err := s.Impl.ShouldBidBasedOnUsage(context.TODO(), args.Req, args.Usage)
+	if err != nil {
+		return err
+	}
+	reply.Resp = result
+	return nil
+}
+
+func (s *ServerRPC) ShouldBig(args ShouldBidRequest, reply *ShouldBidResponse) error {
+	result, err := s.Impl.ShouldBid(context.TODO(), args.Req)
+	if err != nil {
+		return err
+	}
+	reply.Resp = result
+	return nil
+}
+
+func (s *ServerRPC) GetOutputStream(args GetOutputStreamRequest, reply GetOutputStreamResponse) error {
+	result, err := s.Impl.GetOutputStream(context.TODO(), args.ExecutionID, args.WithHistory, args.Follow)
+	if err != nil {
+		return err
+	}
+	// TODO(forrest): [fixme] we expect this method to error as net/rpc does not support streaming response.
+	// NB: gRPC does, and we will need to switch to that if we want streaming.
+	_ = result
+	return nil
+}
+
+func (s *ServerRPC) Run(args RunRequest, reply *RunResponse) error {
+	result, err := s.Impl.Run(context.TODO(), args.ExecutionID, args.ResultsDir, args.Job)
+	if err != nil {
+		return err
+	}
+	reply.Result = result
+	return nil
+}
+
+var _ PluggableExecutor = (*ClientRPC)(nil)
+
+type ClientRPC struct {
```

Comment:
The compute node talks to this client which forwards requests to `ServerRPC` as RPC calls.